iso15118/secc/controller/simulator.py

```diff
@@ -722,30 +722,66 @@ async def get_ac_charge_params_v20(
-            evse_present_active_power_l2=RationalNumber(exponent=3, value=3),
-            evse_present_active_power_l3=RationalNumber(exponent=3, value=3),
+            evse_max_charge_power=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_max_charge_power
+            ),
+            evse_max_charge_power_l2=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_max_charge_power_l2
+            ),
+            evse_max_charge_power_l3=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_max_charge_power_l3
+            ),
+            evse_min_charge_power=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_min_charge_power
+            ),
+            evse_min_charge_power_l2=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_min_charge_power_l2
+            ),
+            evse_min_charge_power_l3=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_min_charge_power_l3
+            ),
+            evse_nominal_frequency=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_nominal_frequency
+            ),
+            max_power_asymmetry=RationalNumber.get_rational_repr(
+                self.evse_data_context.max_power_asymmetry
+            ),
+            evse_power_ramp_limit=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_power_ramp_limit
+            ),
+            evse_present_active_power=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_present_active_power
+            ),
+            evse_present_active_power_l2=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_present_active_power_l2
+            ),
+            evse_present_active_power_l3=RationalNumber.get_rational_repr(
+                self.evse_data_context.evse_present_active_power_l3
+            ),
         )
         if selected_service == ServiceV20.AC:
             return ac_charge_parameter_discovery_res_params
         elif selected_service == ServiceV20.AC_BPT:
             return BPTACChargeParameterDiscoveryResParams(
                 **(ac_charge_parameter_discovery_res_params.dict()),
-                evse_max_discharge_power=RationalNumber(exponent=0, value=3000),
-                evse_max_discharge_power_l2=RationalNumber(exponent=0, value=3000),
-                evse_max_discharge_power_l3=RationalNumber(exponent=0, value=3000),
-                evse_min_discharge_power=RationalNumber(exponent=0, value=300),
-                evse_min_discharge_power_l2=RationalNumber(exponent=0, value=300),
-                evse_min_discharge_power_l3=RationalNumber(exponent=0, value=300),
+                evse_max_discharge_power=RationalNumber.get_rational_repr(
+                    self.evse_data_context.evse_max_discharge_power_l3
```

Comment:
self.evse_data_context.evse_max_discharge_power_l3 -> should be  self.evse_data_context.evse_max_discharge_power

Reply:
Yes, have fixed in now...